tests/themis/themis_secure_session.c

```diff
@@ -668,11 +666,63 @@ static void test_basic_flow_no_transport(void)
     }
 }
 
+static void test_empty_peer_id(void)
+{
+    memcpy(&(client.transport), &transport, sizeof(secure_session_user_callbacks_t));
+    client.transport.user_data = &client;
+
+    client.session =
+        secure_session_create(client.id, 0, client.priv, client.priv_length, &(client.transport));
+
+    testsuite_fail_if(client.session != NULL, "secure_session_create must fail with empty Peer ID");
+}
+
+static void test_invalid_private_key_type(void)
+{
+    memcpy(&(client_rsa.transport), &transport, sizeof(secure_session_user_callbacks_t));
+    client_rsa.transport.user_data = &client_rsa;
```

Comment:
Well, this a weird way to setup a Secure Session – with global variables and memcpy() – but I'll let it pass given that all tests here are like that. It's not nice coding practice because it makes impossible to run the tests in parallel. However, refactoring this is a topic for another day.

Reply:
I guess the easiest way to refactor this is to mark those glogal vars as thread local and then create as many threads as we like, so they will run in parallel. Another question is whether the test framework is ready for such changes. And these tests are really fast, like almost instant, and the only computation-heavy check is `soter rand: NIST STS`, which, btw, only uses one CPU core.

Reply:
You can also just create a new client context and callbacks for each test :) Which is arguably more readable and obvious than thread-local variables. It's more verbose, yeah, but dumb test code is better than smart one, IMO.